### ISSUE TITLE
Broadening Scrapy query causes less results to be returned

### ISSUE THREAD

**Issue by dailenspencer** [scrapy/scrapy#3274]
I am running into an issue and have not yet been able to figure out the root cause. I have the following Scrapy jobs.py file

```
# -*- coding: utf-8 -*-
import scrapy
import re
from scrapy import Request
import csv
import os
from datetime import datetime, timedelta


class JobsSpider(scrapy.Spider):
    
    name = 'jobs'
    allowed_domains = ['craigslist.org']
    present_timestamp = datetime.now()

    
    # This function handles the verification of a craigslist result item top-level data (title, and url)
    # TODO: get rid of this extra argument 'log' that is attached to the extract() response from scrapy
    def verifyTopLevelData (log, title, url, date):

        dataIsValid = True
                
        # ensure post date is not older than 2 days
        post_timestamp = datetime.strptime(date, "%Y-%m-%d %H:%M")
        cutoff_time = datetime.strptime( (datetime.now() - timedelta(days=2)).strftime("%Y-%m-%d 00:00"), "%Y-%m-%d %H:%M" )

        # enforce time-restriction
        if (bool(post_timestamp > cutoff_time) == False):
            dataIsValid = False
        
        return bool(dataIsValid)
    
    # This function handles the building of all craigslist ulrs to extract result list items from
    # TEMPLATE: <rootlocation>.craigstlist.org/search/<search_category>?query=<search_query>
    def gatherCraigsListUrls():
         
        urls = []
                
        # craigslist urls are location based, we build urls based on these root locations
        rootLocations = [
            "bakersfield",
            "chico",
            "fresno",
            "goldcountry",
            "hanford",
            "humboldt",
            "imperial",
            "inlandempire",
            "losangeles",
            "mendocino",
            "merced",
            "modesto",
            "monterey",
            "orangecounty",
            "palmsprings",
            "redding",
            "sacramento",
            "sandiego",
            "sfbay",
            "slo",
            "santabarbara",
            "santamaria",
            "siskiyou",
            "stockton",
            "susanville",
            "ventura",
            "visalia",
            "yubasutter"
        ]

        # craigslist url search categories 
        searchCategories = [
            'ggg',
            'cpg',
            'web',
            'sof',
            'sad',
            'tch',
        ]

        # craigslit url search queries
        # TODO: add regex so we can replace whitespace with '+'
        searchQueries = [
            # web
            'web+developer',
            'web+design',
            'react+developer',
            'angular+developer'
        ]

        # build urls
        for location in rootLocations:
            for category in searchCategories:
                for query in searchQueries:
                    url = 'https://' + location + '.craigslist.org/search/' + category + '?query=' + query
                    urls.append(url)
       
        return urls
  
    start_urls = gatherCraigsListUrls ()

    
    def parse(self, response):
        
        # proxies can sometimes fail. we check to ensure we've landed on the right page. and if not, re-try but using local port
        # if not response.xpath('//a[@class="header-logo"]'):
        #     yield Request(url=response.url, dont_filter=True)

        # grabbing list items from DOM on craigslist result page
        posts = response.xpath('//p[@class="result-info"]') 
        
        # loop through result listings and retrieve top-level data (title, and url)
        for post in posts:

            # top-level data
            post_title = post.xpath('a/text()').extract_first("")
            post_relative_url = post.xpath('a/@href').extract_first()
            post_absolute_url = response.urljoin(post_relative_url)
            
            post_date = post.xpath('time[@class="result-date"]/@datetime').extract_first("")
            push_date = datetime.now().strftime("%Y-%m-%d 00:00"), "%Y-%m-%d %H:%M"

            # verify top-level data and proceed with extracting contents from craigslist post page
            if (self.verifyTopLevelData (post_title, post_absolute_url, post_date)):
                yield Request(post_absolute_url, callback=self.parse_page, meta={'Title':post_title, 'URL': post_absolute_url})
            

        # move to next page (using pagination button on DOM) and repeat "parse" process
        nextpage_relative_url = response.xpath('//a[@class="button next"]/@href').extract_first()
        nextpage_absolute_url = response.urljoin(nextpage_relative_url)

        yield Request(nextpage_absolute_url,callback=self.parse)
        
       
    def parse_page(self, response):

        # proxies can sometimes fail. we check to ensure we've landed on the right page. and if not, re-try but using local port
        # if not response.xpath('//a[@class="header-logo"]'):
        #     yield Request(url=response.url, dont_filter=True)
        
        description = "".join(line for line in response.xpath('//*[@id="postingbody"]/text()').extract())
        response.meta['Description'] = description
        
        yield response.meta
```

If rootLocations is set to an array of just a few items, for example: 
`rootLocations = ["losangeles", "sfbay", "sandieg"]`
then my crawl outputs a bunch of different results. You can see the results here
[results for small query](https://s3.amazonaws.com/craigslist-jobs-app/scrapy-results/2018-05-25_11-39-20_results.json)

However, when I broaden the query to include all craigslist root url locations in california(like in the code example above), I only get back two results. You can see the results here
[results for broad query](https://s3.amazonaws.com/craigslist-jobs-app/scrapy-results/2018-05-25_11-44-50_results.json)

Why would my results shrink when adding more urls to the start_urls array?

**Comment by wRAR** [comment]
The presented code works for me just fine. Does your `gatherCraigsListUrls` return too few starting URLs or why do you get only two results? It's unclear to me.

**Comment by dailenspencer** [comment]
@wRAR when I run the code as is(i.e. with a large set of urls inside gatherCraigsListUrls), then I only receive two results back. However, if I run the code with a smaller set of urls(like the three mentioned before - losangeles, sfbay, and sandiego), then I get more than two results back. How could this be? Are you seeing different results? Do you get more results back when running the larger set of urls?

**Comment by wRAR** [comment]
@dailenspencer but why do you get just two results? Have you looked at the logs and at the requests your spider does? The spider works correctly for me, crawling many URLs and returning some items, but you haven't described clearly how it works for you.

**Comment by dailenspencer** [comment]
@wRAR Okay let me try to explain again. I am getting back *less* results, when I run the crawler on *more* urls.  If i run my crawler on many start_urls, then I only get back two results. But if i run my crawler only on a few of those urls, i get back way more than two results. So my question is - how could that be? I've tried debugging the crawler and cant see any reason why I would get back *less* results, when running the crawler on *more* urls. Does that make sense?

**Comment by wRAR** [comment]
@dailenspencer I cannot help you if you don't provide more details, sorry.

**Comment by dailenspencer** [comment]
@wRAR  
I apologize. I'm not sure how I could be more clear about the issue. Do you understand the issue i'm having though? When the crawler is sent *more* urls, I receive *less* scraped results. When the crawler runs on just a few urls, the outputted json file has over a dozen results. But when the crawler runs on many urls(like 10, 20, 30, 40+ urls), then the outputted json file only has two results! I'm wondering how that could possibly be. Do those details not make sense?

**Comment by dailenspencer** [comment]
@wRAR   I believe `item_scraped_count` is what i'm looking for. I'll check the numbers and see how they compare to one another.  To your note about the logs, as I was looking through the logs, I saw that it was hitting all the urls as expected. In the first scrapy attempt, scrapy hits maybe 20~ supplied urls and returns back the scraped results. In the second run, scrapy hits 100+ supplied urls(including the 20 that were supplied in the first scrapy attempt) and returns back the scraped results. But this time, it returns back less results. I was a bit stuck on how to investigate further after reviewing the logs. Apologies for not being clear :)

**Comment by Gallaecio** [comment]
I’m closing this issue now.

@dailenspencer If you are still stuck on this, please use [StackOverflow](https://stackoverflow.com/tags/scrapy).
